pkg/sqlite/query.go

```diff
@@ -752,3 +780,157 @@ func (s *SQLQuerier) GetCurrentCSVNameForChannel(ctx context.Context, pkgName, c
+	dependencies.type, dependencies.package_name, dependencies.group_name, dependencies.version, dependencies.kind
+	FROM channel_entry
+	INNER JOIN operatorbundle ON operatorbundle.name = channel_entry.operatorbundle_name
+	INNER JOIN dependencies ON dependencies.operatorbundle_name = channel_entry.operatorbundle_name
+	INNER JOIN package ON package.name = channel_entry.package_name
+	GROUP BY channel_entry.package_name, channel_entry.channel_name`
+
+	rows, err := s.db.QueryContext(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	bundles = []*api.Bundle{}
+	bundlesMap := map[string]*api.Bundle{}
+	for rows.Next() {
+		var entryID sql.NullInt64
+		var bundle sql.NullString
+		var bundlePath sql.NullString
+		var minDepth sql.NullInt64
+		var bundleName sql.NullString
+		var pkgName sql.NullString
+		var channelName sql.NullString
+		var replaces sql.NullString
+		var version sql.NullString
+		var skipRange sql.NullString
+		var depType sql.NullString
+		var depPkgName sql.NullString
+		var depGroup sql.NullString
+		var depVersion sql.NullString
+		var depKind sql.NullString
+		if err := rows.Scan(&entryID, &bundle, &bundlePath, &minDepth, &bundleName, &pkgName, &channelName, &replaces, &version, &skipRange, &depType, &depPkgName, &depGroup, &depVersion, &depKind); err != nil {
+			return nil, err
+		}
+
+		bundleKey := fmt.Sprintf("%s/%s/%s", bundleName.String, version.String, bundlePath.String)
+		bundleItem, ok := bundlesMap[bundleKey]
+		if ok {
+			// Create new dependency object
+			dep := &api.Dependency{}
+			dep.Type = depType.String
+			dep.Name = depPkgName.String
+			dep.Group = depGroup.String
+			dep.Version = depVersion.String
+			dep.Kind = depKind.String
+
+			// Add new dependency to the existing list
+			existingDeps := bundleItem.Dependencies
+			existingDeps = append(existingDeps, dep)
+			bundleItem.Dependencies = existingDeps
+		} else {
+			// Create new bundle
+			out := &api.Bundle{}
+			if bundle.Valid && bundle.String != "" {
+				out, err = registry.BundleStringToAPIBundle(bundle.String)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			out.CsvName = bundleName.String
+			out.PackageName = pkgName.String
+			out.ChannelName = channelName.String
+			out.BundlePath = bundlePath.String
+			out.Version = version.String
+			out.SkipRange = skipRange.String
+
+			provided, required, err := s.GetApisForEntry(ctx, entryID.Int64)
+			if err != nil {
+				return nil, err
+			}
+			out.ProvidedApis = provided
+			out.RequiredApis = required
+
+			// Create new dependency and dependency list
+			dep := &api.Dependency{}
+			dependencies := []*api.Dependency{}
+			dep.Type = depType.String
+			dep.Name = depPkgName.String
+			dep.Group = depGroup.String
+			dep.Version = depVersion.String
+			dep.Kind = depKind.String
+			dependencies = append(dependencies, dep)
+			out.Dependencies = dependencies
+
+			bundlesMap[bundleKey] = out
+		}
+	}
+
+	for _, v := range bundlesMap {
```

Comment:
instead of building a list of deps -> converting to a map to dedupe -> converting back to a list, why not build up the set of dependencies as a map to start with, and then do 1 converstion to a list before assigning to out.Dependencies?

Reply:
The map is to dedupe `api.Bundle` as we have a one (bundle) to many (dependency) relationship. Each row has one bundle to one of many dependency. As a result, there are multiple rows of same bundle (one for reach dependency). Multiple bundles can have the same dependency so a map of dependencies is not helpful unless you meant using value as a list of bundles. I'm not sure if I misunderstood something here.